### PR TITLE
Replaced testng with junit4

### PR DESCRIPTION
**Pull request by ifedorenko**
CI says +1 https://bamboo.zion.sonatype.com/browse/NXO-OSSF5-3

### CODE REVIEW

**Comment by jdillon** [comment]
YAY!

**Comment by adreghiciu** [comment]
+1

**Comment by cstamas** [comment]
+1

File: nexus-test/nexus-test-harness-its/src/test/java/org/sonatype/nexus/integrationtests/client/nexus725/Nexus725InitialRestClientIT.java [hunk 1]
```diff
@@ -68,7 +68,7 @@ public void getRepoListTest()
         NexusClient client = this.getConnectedNexusClient();
 
         List<RepositoryListResource> repos = client.getRepositories();
-        Assert.assertTrue( repos.size() > 0, "Expected list of repos to be larger then 0" );
+        Assert.assertTrue( "Expected list of repos to be larger then 0", repos.size() > 0 );
```
Comment (nabcos) [inline nexus-test/nexus-test-harness-its/src/test/java/org/sonatype/nexus/integrationtests/client/nexus725/Nexus725InitialRestClientIT.java:71]: Please use hamcrest instead.
(SCNR :o)

Comment (ifedorenko) [inline nexus-test/nexus-test-harness-its/src/test/java/org/sonatype/nexus/integrationtests/client/nexus725/Nexus725InitialRestClientIT.java:71]: Yes, these do need to be cleaned up and fixed, but they are unrelated to testng->junit4 migration and should be dealt with separately.

<rant>
At least this code asserts something meaningful. There are places where tests assert static string is not null (likely due to mixed parameter order when the tests were originally migrated from junit to testng) and places tests that simply make no sense, like assertNotNull(int) I mentioned yesterday. 
</rant>

**Comment by peterlynch** [comment]
+1

File: nexus-test/nexus-test-harness-its/src/test/java/org/sonatype/nexus/integrationtests/client/nexus725/Nexus725InitialRestClientIT.java (hunk 1 above)
Comment (jdillon) [inline nexus-test/nexus-test-harness-its/src/test/java/org/sonatype/nexus/integrationtests/client/nexus725/Nexus725InitialRestClientIT.java:71]: I agree, it should using hamcrest or fest or something else, but that we shouldn't complicate the killing of testng with rewriting tests wrt to how they assert.

**Pull request merged**